src/core/connection.c

```diff
@@ -7154,6 +7166,11 @@ QuicConnApplyNewSettings(
             Connection->State.FixedBit = (RandomValue % 2);
             Connection->Stats.GreaseBitNegotiated = TRUE;
         }
+
+        if (Connection->Settings.EcnEnabled) {
+            QUIC_PATH* Path = &Connection->Paths[0];
+            Path->EcnValidationState = ECN_VALIDATION_TESTING;
```

Comment:
Should we have ETW logs for every time a path changes ECN validation state? It shouldn't happen that often, right?

Reply:
I can add one for transiting to unknown state. We have logs for capable and failed.

Reply:
I would like a single (manifested) `PathEcnStateChange` event that we could automate processing in WPA if we want, please.

Reply:
What's the point of automating this in WPA? In most cases, the validation succeeds right after the initial packet.

Reply:
When there's actually something interesting happening, the state transition isn't quite useful. What really useful are the counter values in EcnValidationFailure log.

Reply:
As discussed over IM, should anything be promoted to ETW?

Reply:
Added connection rundown ETW event and ecn failure ETW event.